src/libstd/collections/hash/map.rs

```diff
@@ -410,6 +422,16 @@ fn search_hashed<K, V, M, F>(table: M, hash: SafeHash, mut is_match: F) -> Inter
         return InternalEntry::TableIsEmpty;
     }
 
+    search_hashed_body(table, hash, is_match)
+}
+
+/// The body of the search_hashed[_nonempty] functions
+#[inline]
+fn search_hashed_body<K, V, M, F>(table: M, hash: SafeHash, mut is_match: F)
```

Comment:
Can we make this `search_hashed_nonempty`? Avoiding the extra function.

Reply:
Yes, you're right